sklearn/utils/testing.py

```diff
@@ -796,6 +796,20 @@ def _get_args(function, varargs=False):
         return args
 
 
+def _get_parent_args(Estimator):
+    """Helper to get list of parents properties"""
+    # NOTE this works only in python3.5
+    if sys.version_info < (3, 5):
+        NotImplementedError("_get_parent_args is not available "
```

Comment:
there is no `inspect.signatures` in python 2.  `inspect.getargspec` is the equivalent. the problem is  when `__init__` method is inherited directly from a built-in, `getargspec` fails